JoinColony/colonyCDapp
# Port over member reputation

## Description

This PR ports over member reputation. This includes:
 - Creating a new `getUserReputation` lambda function that generates a GraphQL query
 - Porting over and refactoring `MemberReputation` component
 - Creating a `useUserReputation` hook to encapsulate reputation fetching logic
 - In response to Raul's comment, this PR also adds a functions-specific eslint config to disable a conflicting eslint rule

## To test
As there's no reputation in any of the colonies yet, you'll see the following:
<img width="376" alt="Screenshot 2022-11-04 at 21 19 36" src="https://user-images.githubusercontent.com/112586815/200076768-e798b4e8-b9b5-4271-bd80-93e528ce2a18.png">

You can edit the lambda function to return ~13.81% reputation by applying the following patch:
```
diff --git a/amplify/backend/function/getUserReputation/src/index.js b/amplify/backend/function/getUserReputation/src/index.js
index b725cd6..b5babf2 100644
--- a/amplify/backend/function/getUserReputation/src/index.js
+++ b/amplify/backend/function/getUserReputation/src/index.js
@@ -3,6 +3,7 @@ const { getColonyNetworkClient, Network } = require('@colony/colony-js');
 const {
   providers,
   utils: { Logger },
+  constants: { AddressZero },
 } = require('ethers');
 
 Logger.setLogLevel(Logger.levels.ERROR);
@@ -35,16 +36,20 @@ exports.handler = async (event) => {
 
   const { skillId } = await colonyClient.getDomain(domainId ?? ROOT_DOMAIN_ID);
 
-  try {
-    const { reputationAmount } = await colonyClient.getReputationWithoutProofs(
-      skillId,
-      walletAddress,
-      rootHash,
-    );
+  return walletAddress === AddressZero
+    ? '11904644120813032491751554'
+    : '1644530790622401088627970';
 
-    return reputationAmount.toString();
-    // eslint-disable-next-line no-empty
-  } catch {}
+  // try {
+  //   const { reputationAmount } = await colonyClient.getReputationWithoutProofs(
+  //     skillId,
+  //     walletAddress,
+  //     rootHash,
+  //   );
 
-  return null;
+  //   return reputationAmount.toString();
+  //   // eslint-disable-next-line no-empty
+  // } catch {}
+
+  // return null;
 };

```

Once you refresh, you should see the following:
<img width="373" alt="Screenshot 2022-11-04 at 21 19 45" src="https://user-images.githubusercontent.com/112586815/200076938-30eb819c-30dd-4e71-9eb4-5aee10ac305b.png">

## TODO

- [x] Lambda function
- [x] MemberReputation component

Resolves #62
